gems/aws-sdk-core/lib/aws-sdk-core/plugins/endpoint_discovery.rb

```diff
@@ -151,8 +155,12 @@ def _discover_endpoint(ctx, required)
 
       def self.resolve_endpoint_discovery(cfg)
         env = ENV['AWS_ENABLE_ENDPOINT_DISCOVERY']
+        default = cfg.api.operations.any? do |_, o|
+          o.endpoint_discovery && o.endpoint_discovery['required']
+        end
```

Comment:
Wasn't this already determined above in `endpoint_discovery_required`?

Reply:
This is the confusing part of using a dynamic default (ie, the `do |cfg| resolve_endpoint_discovery(cfg) end`.  [Plugin#add_option](https://github.com/aws/aws-sdk-ruby/blob/master/gems/aws-sdk-core/lib/seahorse/client/plugin.rb#L9) and [Configuration#add_option](https://github.com/aws/aws-sdk-ruby/blob/master/gems/aws-sdk-core/lib/seahorse/client/configuration.rb#L106) let us either have a static default or a dynamic (block) default.  The dynamic default is called by the Configuration::DefaultResolver during build! (which is called during [Client::Base#initiailize](https://github.com/aws/aws-sdk-ruby/blob/master/gems/aws-sdk-core/lib/seahorse/client/base.rb#L19)) and doesn't have access to the doc_default that is resolved during code gen...

I don't like the disconnect between the doc_default and the behavior of the dynamic default - I'm open to ideas on how to connect them, but I don't have any great ones.